src/silverlabnwb/subsample_nwb.py

```diff
@@ -189,31 +188,37 @@ def cycles_per_trial(nwb):
     trial. Currently looks at the first imaging timeseries in
     the first trial, and assumes they're all the same.
     """
-    trial1 = nwb['/epochs/trial_0001']
-    for ts_name in trial1:
-        ts = trial1[ts_name]
-        is_image_series = ts['timeseries/pixel_time_offsets'] is not None
-        if is_image_series:
-            return ts['count'].value
-    else:
-        raise ValueError('No imaging timeseries found')
+    n_all_trials = len(nwb['/intervals/trials/id'])
+    return np.int(nwb['acquisition/ROI_001_Red/timestamps'].shape[0] / n_all_trials)
 
 
 def copy_tdms(nwb, in_path, out_path, nrois):
-    num_all_rois = nwb['/processing/Acquired_ROIs/roi_spec'].shape[0]
+    all_rois = [nwb['/acquisition/'+key] for key in nwb['/acquisition/'].keys() if key.startswith('ROI') and key.endswith('Red')]
+    num_all_rois = len(all_rois)
+    all_roi_dimensions_pixels = np.array([roi['dimension'] for roi in all_rois])
     print('Copying {} of {} ROIs from {} to {}'.format(
         nrois, num_all_rois, in_path, out_path))
     in_tdms = nptdms.TdmsFile(in_path)
     group_name = 'Functional Imaging Data'
     with nptdms.TdmsWriter(out_path) as out_tdms:
-        root, group = in_tdms.object(), in_tdms.object(group_name)
-        out_tdms.write_segment([root, group])
+        group = in_tdms[group_name]
+        out_tdms.write_segment([group])
         for ch, channel in {'0': 'Red', '1': 'Green'}.items():
             ch_name = 'Channel {} Data'.format(ch)
-            ch_obj = in_tdms.object(group_name, ch_name)
-            shape = (cycles_per_trial(nwb), num_all_rois, -1)
-            ch_data = ch_obj.data.reshape(shape)
-            subset = ch_data[:, :nrois, :].reshape(-1)
+            ch_obj = in_tdms[group_name][ch_name]
+            # The number of pixels for each ROI for one cycle, and for all ROIs
+            all_roi_pixels = all_roi_dimensions_pixels.prod(axis=1)
+            total_pixels = all_roi_pixels.sum()
+            # How many pixels do the previous ROIs take up within a cycle's data?
+            # We'll need this to see where to start reading for a particular ROI.
+            previous_pixels = np.concatenate(([0], all_roi_pixels[:-1].cumsum()))
+            subset = np.array(())
+            for roi_index, roi_to_keep in enumerate(all_rois[0:nrois]):
+                starts = np.arange(cycles_per_trial(nwb)) * total_pixels + previous_pixels[roi_index]
+                stops = starts + all_roi_pixels[roi_index]
+                inds = np.array([np.arange(start, stop) for (start, stop) in zip(starts, stops)])
+                ch_data = ch_obj.data[inds].flatten()
+                subset = np.concatenate((subset, ch_data))
```

Comment:
So this gets all the data for one ROI across all cycles, then appends the data for the next ROI for all cycles etc,  but the original TDMS is structured the other way around: it has data for all ROIs for the first cycle, then all ROIs for the second cycle etc. This may be why we the sampled file is not consistent with the original!